src/IncludedCommands/Resources/PersistentBans.luau

```diff
@@ -55,69 +56,71 @@ function PersistentBans.Initialize(self: PersistentBans): ()
+        warn(`Unable to use RobloxBanApi as Players.BanningEnabled is set to false. Persistent bans disabled.`)
+        self.Initialized = true
+        self.Enabled = false
+        return
+    elseif not Worked then
         self.Initialized = true
+        warn(`Persistent Bans failed to load because {ErrorMessage}`)
         return
     end
-    
-    --Connect kicking players.
-    Players.PlayerRemoving:Connect(function(Player: Player)
-        if not self.PlayerUpdateEvents[Player] then return end
-        self.PlayerUpdateEvents[Player]:Disconnect()
-        self.PlayerUpdateEvents[Player] = nil
-    end)
-    Players.PlayerAdded:Connect(function(Player: Player)
-        self:ConnectPlayer(Player)
+
+    xpcall(function()
+        self.LogsDataStore = NexusDataStore:GetDataStore("NexusAdmin_Persistence", "PersistentBan_Logs")
+    end, function(ErrorMessage)
+        warn(`Failed to get persistent ban logs because {ErrorMessage}`)
     end)
-    for _, Player in Players:GetPlayers() do
-        self:ConnectPlayer(Player)
-    end
+
+    self:TransferLegacy()
+
     self.Initialized = true
 end
 
 --[[
-Connects a player.
+Transfer legacy persistent bans
 --]]
-function PersistentBans.ConnectPlayer(self: PersistentBans, Player: Player): ()
-    if not self.BansDataStore then return end
-    self.PlayerUpdateEvents[Player] = self.BansDataStore:OnUpdate(tostring(Player.UserId), function()
-        self:KickPlayer(Player)
-    end)
-    self:KickPlayer(Player)
-end
+function PersistentBans.TransferLegacy(self: PersistentBans): ()
+    --Get the DataStore.
+    local BansDataStore
 
---[[
-Kicks a player if they are banned.
---]]
-function PersistentBans.KickPlayer(self: PersistentBans, Player: Player): ()
-    if not self.BansDataStore then return end
-    task.spawn(function()
-        --Return if the player is an admin.
-        self.Api.Authorization:YieldForAdminLevel(Player)
-        if self.Api.Authorization:IsPlayerAuthorized(Player, 0) then
-            return
-        end
+    local Worked, ErrorMessage = pcall(function()
+        BansDataStore = NexusDataStore:GetDataStore("NexusAdmin_Persistence", "PercistentBans")
+    end)
+    if not Worked then
+        warn(`Failed to get persistent bans because {ErrorMessage}`)
+        warn(`Persistent bans legacy transfer aborted`)
+        return
+    end
 
-        --Kick the player.
-        local BanMessage = self.BansDataStore:Get(tostring(Player.UserId))
-        if BanMessage == true then
-            Player:Kick()
-        elseif BanMessage then
-            Player:Kick(BanMessage)
+    --Call :BanId individually for each legacy ban, assuming no two bans have the same reason.
+    for _, Id in BansDataStore:ListKeys() do
+        local Reason = BansDataStore:Get(Id)
+        local Worked, ErrorMessage = self:BanId(tonumber(Id), Reason, -2, "NexusAdmin-Legacy pban transfer")
```

Comment:
`tonumber(Id)` returns `number?`. `tonumber(Id) :: number` will silence this typing issue.